src/pokete/classes/difficulty.py

```
@@ -33,14 +38,78 @@ def __init__(self):
+                del self.roadblocks[trainer_name]
+        else:
+            # Register loss/roadblock
+            if trainer_name:
+                self.register_loss(trainer_name)
+
+        self.update_performance(rating, variety_score, efficiency_score)
+
+    def register_loss(self, trainer_name):
+        """Tracks losses against specific trainers."""
+        self.roadblocks[trainer_name] = self.roadblocks.get(trainer_name, 0) + 1
+        logging.info(f"[DDA] Roadblock registered: {trainer_name} (Count: {self.roadblocks[trainer_name]})")
```

Ruff (0.14.14)

[warning] `info()` call on root logger

(LOG015)